tests/unit/synapseclient/unit_test_tables.py

```diff
@@ -119,6 +121,106 @@ def test_cast_values__list_type():
     ]
 
 
+def test_convert_df_date_cols_to_datetime_empty_df():
+    test_df = pd.DataFrame()
+    date_columns = []
+    test_df_output = _convert_df_date_cols_to_datetime(test_df, date_columns)
+    assert test_df_output.empty
+
+
+def test_convert_df_date_cols_to_datetime_invalid_epoch_time():
+    test_df = pd.DataFrame(
+        {
+            "epoch_time1": [
+                "1107234000000",
+                "invalid epoch time",
+                "invalid epoch time",
+                "invalid epoch time",
+                "invalid epoch time",
+            ]
+        }
+    )
+    date_columns = ["epoch_time1"]
+    with pytest.raises(ValueError):
+        test_df_output = _convert_df_date_cols_to_datetime(test_df, date_columns)
+
+
+def test_convert_df_date_cols_to_datetime_invalid_date_columns():
+    test_df = pd.DataFrame(
+        {
+            "epoch_time1": [
+                "1107234000000",
+                "1107320400000",
+                "1107406800000",
+                "1107493200000",
+                "1107579600000",
+            ]
+        }
+    )
+    date_columns = ["invalid date column"]
+    with pytest.raises(ValueError):
+        test_df_output = _convert_df_date_cols_to_datetime(test_df, date_columns)
+
+
+def test_convert_df_date_cols_to_datetime():
+    # construct test dataframe and get date columns
+    test_df = pd.DataFrame(
+        {
+            "row": [1, 2, 3, 4, 5],
+            "epoch_time1": [
+                "1107234000000",
+                "1107320400000",
+                "1107406800000",
+                "1107493200000",
+                "1107579600000",
+            ],
+            "epoch_time2": [
+                "1107234000000",
+                "1107320400000",
+                "1107406800000",
+                "1107493200000",
+                "1107579600000",
+            ],
+            "string": ["str1", "str2", "str3", "str4", "str5"],
+        }
+    )
+    date_columns = ["epoch_time1", "epoch_time2"]
+
+    # construct expected date dataframe
+    expected_date_df = pd.DataFrame(
+        {
+            "epoch_time1": [
+                "2005-02-01 05:00:00+00:00",
+                "2005-02-02 05:00:00+00:00",
+                "2005-02-03 05:00:00+00:00",
+                "2005-02-04 05:00:00+00:00",
+                "2005-02-05 05:00:00+00:00",
+            ],
+            "epoch_time2": [
+                "2005-02-01 05:00:00+00:00",
+                "2005-02-02 05:00:00+00:00",
+                "2005-02-03 05:00:00+00:00",
+                "2005-02-04 05:00:00+00:00",
+                "2005-02-05 05:00:00+00:00",
+            ],
+            "string": ["str1", "str2", "str3", "str4", "str5"],
+        }
+    )
+    expected_date_df[["epoch_time1", "epoch_time2"]] = expected_date_df[
+        ["epoch_time1", "epoch_time2"]
+    ].astype("datetime64[ns, UTC]")
+
+    # convert epoch time to date time
+    test_df2 = _convert_df_date_cols_to_datetime(test_df, date_columns)
```

Comment:
I think you can just do:

```
assert_frame_equal(test_df2_dates, expected_date_df)
```

Reply:
Yes that's a good point. Let me simplify the code further. Thanks for the suggestion.